mongo_connector/doc_managers/nodes_and_relationships_updater.py

```diff
@@ -47,6 +57,10 @@ def update_relationship(self, document, root_type, doc_type, doc_id):
     self.statements_with_params.append(builder.query_nodes)
     self.statements_with_params.append(builder.relationships_query)
 
+  def remove_legacy_nodes(self, doc_id, doc_type):
+    params_dict = {"doc_id": doc_id}
+    statement = "MATCH (d:Document:{doc_type}) WHERE d._id={{doc_id}} OPTIONAL MATCH (d)-[r]-(c) DELETE r, c".format(doc_type=doc_type)
```

Comment:
If node `(c)` has relationships other than `r`, won't this query fail? I understand that we're leaving some of this to be resolved in #34 but can we avoid a possible failing query with something like:

    MATCH (d:Document) WHERE d._id={doc_id} 
    OPTIONAL MATCH (d)-[r]-(c)
    DELETE r
    WITH d,c
    OPTIONAL MATCH (c)-[s]-()
    WITH d,c,s,
    CASE
        WHEN s IS NULL THEN c
        ELSE NULL
    END AS n
    DELETE n

By using the `CASE` clause here we can check if `(c)` has additional relationships and only delete `(c)` if it has no other relationships. Might want to check if my logic is right, but I think this will work.